Description: FileChooser portal: send window id in hex
 We send window id in decimal, however, it is expected to be send in hex.
 This causes a mismatch and makes portal dialog to show in background.
Origin: upstream, https://code.qt.io/cgit/qt/qtbase.git/commit/?id=acaabc9108dfe75530960cf8e3ec4f3602cd82e0
Last-Update: 2021-11-04
Index: qtbase-opensource-src/src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportalfiledialog.cpp
===================================================================
--- qtbase-opensource-src.orig/src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportalfiledialog.cpp
+++ qtbase-opensource-src/src/plugins/platformthemes/xdgdesktopportal/qxdgdesktopportalfiledialog.cpp
@@ -185,7 +185,7 @@ void QXdgDesktopPortalFileDialog::openPo
                                                           QLatin1String("/org/freedesktop/portal/desktop"),
                                                           QLatin1String("org.freedesktop.portal.FileChooser"),
                                                           d->saveFile ? QLatin1String("SaveFile") : QLatin1String("OpenFile"));
-    QString parentWindowId = QLatin1String("x11:") + QString::number(d->winId);
+    QString parentWindowId = QLatin1String("x11:") + QString::number(d->winId, 16);
 
     QVariantMap options;
     if (!d->acceptLabel.isEmpty())
